### PR TITLE
Pubnative: Fix Shared Memory Overwriting

### PR DESCRIPTION
**Pull request by guscarreon**
Copy of reference field `Banner` needed to be made inside `adapters/pubnative/pubnative.go` in order to avoid a data race on shared memory. Authors of source code should be notified and encourage to make any changes to this PR if they disagree with the proposed approach.

### CODE REVIEW

**Comment by SyntaxNode** [comment]
Please contact Pubnative using their bidder info email address to inform them of this change.

File: adapters/pubnative/pubnative.go [hunk 1]
```diff
@@ -94,29 +94,36 @@ func convertImpression(imp *openrtb.Imp) error {
...
 		if len(banner.Format) > 0 {
 			f := banner.Format[0]
-			banner.W = &f.W
-			banner.H = &f.H
+
+			bannerCopy := *banner
+			bannerCopy.W = new(uint64)
+			*bannerCopy.W = f.W
+			bannerCopy.H = new(uint64)
+			*bannerCopy.H = f.H
```
Comment (SyntaxNode) [inline adapters/pubnative/pubnative.go:117]: Please use the openrtb.Uint64Ptr() helper method. It's a bit cleaner:

```
bannerCopy.W = openrtb.Uint64Ptr(f.W)
bannerCopy.H = openrtb.Uint64Ptr(f.H)
```

**Comment by guscarreon** [comment]
@fhartwig,
Recently coded pull request #1756 is meant to enable Prebid Server to find data race conditions in adapter bidder source code. According to this newly coded feature, your adapter might be writing to shared memory and this PR implements an approach to prevent it. Please let us know whether or not you agree with the changes proposed in this pull request or if you rather want to code the fix yourself. 

When it comes to adapters' source code improvements, we usually wait a week for owners to come back to us. If we get no feedback, we will go ahead to review and merge this pull request ourselves.

We'd love to hear your take on this issue.

File: adapters/pubnative/pubnative.go [hunk 2]
```diff
@@ -94,29 +94,35 @@ func convertImpression(imp *openrtb.Imp) error {
...
+
+			bannerCopy := *banner
+
+			bannerCopy.W = openrtb.Uint64Ptr(f.W)
+			bannerCopy.H = openrtb.Uint64Ptr(f.H)
+
+			return &bannerCopy, nil
 		} else {
-			return &errortypes.BadInput{
+			return banner, &errortypes.BadInput{
```
Comment (SyntaxNode) [inline adapters/pubnative/pubnative.go:120]: Should this return a `nil` banner if there's an error?

Comment (fhartwig) [reply adapters/pubnative/pubnative.go:120]: I'd probably have gone with nil, but either way seems fine to me.

**Review by fhartwig — DISMISSED** [review]
This looks good to me, thanks for fixing it!

**Comment by guscarreon** [comment]
> This looks good to me, thanks for fixing it!

Thanks for your review  @fhartwig

**Review by hhhjort — APPROVED** [review]
LGTM

**Review by SyntaxNode — APPROVED** [review]

**Review by hhhjort — APPROVED** [review]

**Pull request merged**